src/Npgsql/NpgsqlBatchCommand.cs

```diff
@@ -38,6 +38,33 @@ public override string CommandText
     /// <inheritdoc cref="DbBatchCommand.Parameters"/>
     public new NpgsqlParameterCollection Parameters { get; } = new();
 
+#pragma warning disable CA1822 // Mark members as static
+
+#if NET8_0_OR_GREATER
+    /// <inheritdoc/>
+    public override
+#else
+    /// <summary>
+    /// Creates a new instance of an <see cref="System.Data.Common.DbParameter"/> object.
+    /// </summary>
+    /// <returns>A <see cref="System.Data.Common.DbParameter"/> object.</returns>
+    public
+#endif
+    DbParameter CreateParameter() => new NpgsqlParameter();
```

Comment:
Can you make this return NpgsqlParameter in the 8.0 version? 

The idea behind these methods being added as public virtual (as opposed to the ado.net protected virtual pattern) is to be able to use covariant return types.

Reply:
Done.